src/include/lkl/lkl_util.h

```diff
@@ -3,13 +3,66 @@
 
 #include <openenclave/enclave.h>
 
-#define SGXLKL_LKL_SYSCALL 1
-#define SGXLKL_INTERNAL_SYSCALL 3
-#define SGXLKL_IGNORED_SYSCALL 4
-#define SGXLKL_UNSUPPORTED_SYSCALL 5
-#define SGXLKL_REDIRECT_SYSCALL 6
+/**
+ * The mechanism used to implement a specified system call.
+ */
+typedef enum
+{
+    /**
+     * System calls that are implemented by LKL as a normal Linux system call.
+     */
+    SGXLKL_LKL_SYSCALL = 1,
+    /**
+     * System calls that are implemented using SGX-LKL custom code paths that
+     * bypass the Linux kernel.  Eventually, these will all be replaced.
+     */
+    SGXLKL_INTERNAL_SYSCALL = 3,
+    /**
+     * System calls that are not possible to implement in an SGX environment but
+     * software that expects it to work can act as if it does.  For example,
+     * `mlock` is ignored because memory is all locked by the SGX environment.
+     */
+    SGXLKL_IGNORED_SYSCALL = 4,
+    /**
+     * System calls that are not supported in the SGX
+     * environment at all and which callers may handle.
+     */
+    SGXLKL_UNSUPPORTED_SYSCALL = 5,
+    /**
+     * System calls that were invoked using the normal
+     * architecture's system call numbers and
+     * redirected.
+     */
+    SGXLKL_REDIRECT_SYSCALL = 6,
+} sgxlkl_syscall_kind;
 
-void __sgxlkl_log_syscall(int type, long n, long res, int params_len, ...);
+/**
+ * Logs a trace message for the specified syscall.  The type argument
+ * identifies how the system call is implemented, `n` gives the number of
+ * parameters, `res` the result value. There are `params_len` variadic
+ * parameters (0 to 6) that give the arguments for the system call.
+ *
+ * This function returns `res`, so that it can be tail called on the return
+ * path.
+ */
+#if DEBUG
```

Comment:
We're having a discussing if this kind of tracing should only be possible for the DEBUG build. I believe that the emerging opinion is that it should also be possible for the other build types.  @letmaik @wintersteiger can you confirm?

Reply:
I've moved it under a separate flag, which is currently defined to be the same as `DEBUG` but can easily be changed later.

Reply:
Yes, confirmed, new flag is fine too.